src/diffusers/pipelines/consistency_models/pipeline_consistency_decoder.py

```diff
@@ -0,0 +1,135 @@
+    r"""
+    Decoding pipeline using the consistency decoder used with DALL-E 3.
+
+    This model inherits from [`DiffusionPipeline`]. Check the superclass documentation for the generic methods
+    implemented for all pipelines (downloading, saving, running on a particular device, etc.).
+
+    Args:
+        unet ([`UNet2DModel`]):
+            A `UNet2DModel` to denoise the encoded image latents.
+        scheduler ([`SchedulerMixin`]):
+            A scheduler to be used in combination with `unet` to denoise the encoded image latents. Currently only
+            compatible with [`ConsistencyDecoderScheduler`].
+    """
+
+    model_cpu_offload_seq = "unet"
+
+    def __init__(self, unet: UNet2DModel, scheduler: ConsistencyDecoderScheduler) -> None:
+        super().__init__()
+
+        self.register_modules(unet=unet, scheduler=scheduler)
+
+        self.image_processor = VaeImageProcessor()
+
+    @torch.no_grad()
+    @replace_example_docstring(EXAMPLE_DOC_STRING)
+    def __call__(
+        self,
+        z: torch.Tensor,
+        generator: Optional[torch.Generator] = None,
+        output_type: Optional[str] = "pil",
+        num_inference_steps: int = 2,
+        return_dict: bool = True,
+    ):
+        r"""
+        Args:
+            z (`torch.Tensor`):
+                Latents which are typically precomputed from a reverse diffusion denoiser or an encoder.
+            num_inference_steps (`int`, *optional*, defaults to 2):
+                The number of denoising steps. More denoising steps usually lead to a higher quality image at the
+                expense of slower inference. Currently it doesn't support a quantity greater than 2.
+            generator (`torch.Generator`, *optional*):
+                A [`torch.Generator`](https://pytorch.org/docs/stable/generated/torch.Generator.html) to make
+                generation deterministic.
+            output_type (`str`, *optional*, defaults to `"pil"`):
+                The output format of the generated image. Choose between `PIL.Image` or `np.array`.
+            return_dict (`bool`, *optional*, defaults to `True`):
+                Whether or not to return a [`~pipelines.ImagePipelineOutput`] instead of a plain tuple.
+
+        Examples:
+
+        Returns:
+            [`~pipelines.ImagePipelineOutput`] or `tuple`:
+                If `return_dict` is `True`, [`~pipelines.ImagePipelineOutput`] is returned, otherwise a `tuple` is
+                returned where the first element is a list with the generated images.
+        """
+        means = torch.tensor([0.38862467, 0.02253063, 0.07381133, -0.0171294], device=z.device, dtype=z.dtype)[
+            None, :, None, None
+        ]
+        stds = torch.tensor([0.9654121, 1.0440036, 0.76147926, 0.77022034], device=z.device, dtype=z.dtype)[
+            None, :, None, None
+        ]
+        z = (z - means) / stds
+        z = F.interpolate(z, mode="nearest", scale_factor=8)
```

Comment:
Does scale factor affect output quality? Should it be passed in as a param to `__call__`?

Reply:
scale factor is required to be the same as the downsampling factor of the encoding